### PR TITLE
fix segment info in Kafka indexing service docs

### PR DESCRIPTION
**Pull request by pjain1**
Fixes https://github.com/druid-io/druid/issues/5384

### CODE REVIEW

File: docs/content/development/extensions-core/kafka-ingestion.md [hunk 1]
```diff
@@ -311,35 +311,17 @@ In this way, configuration changes can be applied without requiring any pause in
...
-will not be sufficient to keep the number of segments at an optimal level. It is recommended that scheduled re-indexing
-tasks be run to merge segments together into new segments of an ideal size (in the range of ~500-700 MB per segment).
+Each Kafka Indexing Task puts events consumed from Kafka partitions assigned to it in a single segment for each segment
+granular interval. Kafka Indexing Task also does incremental hand-offs which means that all the segments created by a
+task will not be held up till the task duration is over. As soon as maxRowsPerSegment limit is hit, all the segments held
+by the task at that point in time will be handed-off and new set of segments will be created for further events.
+This means that the task can run for longer durations of time without accumulating old segments locally on Middle Manager
+nodes and it is encouraged to do so.
+
+Kafka Indexing Service may still produce some small segments. Lets say the task duration is 4 hours, segment granulairty
```
Comment (gianm) [inline docs/content/development/extensions-core/kafka-ingestion.md]: granularity (spelling)

**Comment by pjain1** [comment]
@gianm does this look ok now ?

**Comment by gianm** [comment]
Going to merge this without CI completing since there is no way it will complain about a doc change.

**Comment by gianm** [comment]
Backport in #5393

**Pull request merged**